apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -102,6 +112,84 @@ export const pageRouter = router({
         ...siteMeta,
       }
     }),
+
+  reorderBlock: protectedProcedure
+    .input(reorderBlobSchema)
+    .mutation(async ({ input: { pageId, from, to, blocks, siteId }, ctx }) => {
+      // NOTE: we have to check against the page's content that we retrieve from db
+      // we adopt a strict check such that we allow the update iff the checksum is the same
+      const fullPage = await getFullPageById({ resourceId: pageId, siteId })
+
+      if (!fullPage?.content) {
+        // TODO: we should probably ping on call
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "Unable to load content for the requested page, please contact Isomer Support",
+        })
+      }
+
+      const actualBlocks = fullPage.content.content
+
+      if (!isEqual(blocks, actualBlocks)) {
+        throw new TRPCError({
+          code: "CONFLICT",
+          message:
+            "Someone on your team has changed this page, refresh the page and try again",
+        })
+      }
+
+      if (
+        from >= actualBlocks.length ||
+        to >= actualBlocks.length ||
+        from < 0 ||
+        to < 0
+      ) {
+        // NOTE: If this happens, this indicates that either our dnd libary on our frontend has a
+        // bug or someone is trying to mess with our frontend
+        throw new TRPCError({ code: "UNPROCESSABLE_CONTENT" })
+      }
+
+      const [movedBlock] = actualBlocks.splice(from, 1)
+      if (!movedBlock) return blocks
+      if (!fullPage.draftBlobId || !fullPage.mainBlobId) {
```

Comment:
i think this is a bug right. a page may have either a `draftBlobId` or a `mainBlobId`. One or the other may be null. This conditional will be wrongly truthy.

```suggestion
      if (!fullPage.draftBlobId && !fullPage.mainBlobId) {
```

Reply:
yes you're right omg LEMME UPDATE